pkg/controllers/user/alert/watcher/workload.go

```diff
@@ -92,7 +96,13 @@ func (w *WorkloadWatcher) watchRule() error {
 				continue
 			}
 			for _, wl := range wls {
-				w.checkWorkloadCondition(wl, alert)
+				ns, err := w.namespaceLister.Get("", wl.Namespace)
```

Comment:
I feel like we can optimize it a little. There are a couple of options:

1) Either store namespace who's project has already been validated, in internal map, so we don't have to a) fetch it from cache again b) check project annotation on line 103

or 

2) Only fetch workloads from the alert's project's namespace(s) on line 93.

I like option 2) better because today we get all workloads from all namespaces, and we run on line 93, and then we run the check for every one of them.